Add benchmark to test Kafka Sink (Avro/Debezium) performance (#6649)
MaterializeInc/materialize: test/bench/kafka-sink-avro-debezium/README.md

# Kafka Avro Debezium Sink Benchmark

> This benchmark tests our ability to write messages out to Kafka. In it's
> current configuration, it uses a topic with only one partition as input and a
> topic with only one partition as output. We use only one input topic because
> that is enough to saturate writing. We use only one output topic because the
> current Kafka sink will only ever write from on timely worker.

Reviewer: nit:
```suggestion
This benchmark tests our ability to write messages out to Kafka. In its
current configuration, it uses a topic with only one partition as input and a
topic with only one partition as output. We use only one input topic because
that is enough to saturate writing. We use only one output topic because the
current Kafka sink will only ever write from one timely worker.
```

> ## Running the Avro Upsert Test

Reviewer 2: Nit: s/avro upsert/avro debezium sink.

Author: 👍

> To run the spec benchmark run the following command:

Reviewer 2: I checked out your PR and I can confirm that this also works with `mzbench`, if you want to document that as well:

```sh
[chris-work:~/materialize/materialize:(xxxx-add-sink-benchmark[|])] ./bin/mzbench -w --size medium kafka-sink-avro-debezium
git_revision,num_workers,iteration,seconds_taken,rows_per_second,grafana_url
None,24,0,297,67340,http://localhost:3000/d/materialize-overview/materialize-overview?from=1620153866000&to=1620154225000&tz=UTC
```

Author: I didn't document it here because `avro-upsert` also doesn't mention `mzbench`. I could add a line about `mzbench` in both of those benchmarks, WDYT?

Reviewer 2: Sounds great to me! Thanks!
